include/deal.II/base/smartpointer.h

```diff
@@ -388,6 +395,15 @@ SmartPointer<T, P>::operator->() const
 
 
 
+template <typename T, typename P>
+inline T *
+SmartPointer<T, P>::operator->() const
+{
+  return this->get();
```

Comment:
@masterleinad  It is used here. But yes I want to use it in this place:

https://github.com/dealii/dealii/blob/691a748cc129532a499bc5057b06908b7227e518/include/deal.II/non_matching/mapping_info.h#L1217